packages/core/src/Stores/client-store.js

```diff
@@ -2263,12 +2264,15 @@ export default class ClientStore extends BaseStore {
             this.setIsLoggingIn(true);
 
             const redirect_url = sessionStorage.getItem('redirect_url');
+            const is_next_wallet_enabled = localStorage.getObject('FeatureFlagsStore')?.data?.next_wallet;
+
+            const wallet_redirect_url = is_next_wallet_enabled ? routes.wallets : routes.traders_hub;
 
             if (
                 (redirect_url?.endsWith('/') || redirect_url?.endsWith(routes.bot)) &&
-                (isTestLink() || isProduction() || isLocal() || isStaging())
+                (isTestLink() || isProduction() || isLocal() || isStaging() || isTestDerivApp())
             ) {
-                window.history.replaceState({}, document.title, '/appstore/traders-hub');
+                window.history.replaceState({}, document.title, wallet_redirect_url);
```

Comment:
kind of scary while reading - suggests me that it always redirect to wallets no matter the feature flag, purely due to variable name :D